pkgs/development/tools/build-managers/waf/default.nix

```diff
@@ -1,5 +1,12 @@
-{ stdenv, fetchFromGitLab, fetchpatch, python, ensureNewerSourcesForZipFilesHook }:
-
+{ stdenv, fetchFromGitLab, fetchpatch, python, ensureNewerSourcesForZipFilesHook
+# optional list of extra waf tools, e.g. `[ "doxygen" "pytest" ]`
+, withTools ? [ ]
+}:
+let
+  wafToolsArg = with stdenv.lib.strings;
+    if builtins.length withTools > 0 then "--tools=\"${concatStringsSep "," withTools}\""
```

Comment:
I was thinking about exactly this kind of implementation. There is a small issue with this one though, it is that it prevents user from setting "--tools" to just empty list (I'm not sure what would be the use case for that, but just in case someone needs that). The default value of `withTools` could be changed to `null` and then checked using `isNull`.

You use `with stdenv.lib.strings;`, but only take `concatStringsSep` from it. Perhaps you also wanted to use `optionalString` instead of `if`?

Reply:
Thanks for looking out for me, I'm not really fluent in Nix yet. I've updated the commit to reflect this.